jsk_fetch_robot/jsk_fetch_startup/euslisp/navigation-utils.l

```diff
@@ -132,16 +132,40 @@
     (return-from undock nil))
   (send (send *undock-action* :get-result) :undocked))
 
+(defun wait-until-is-charging (&key timeout 10)
+  """
+  Wait for updating battery charging state especially when robots dock.
+  Args:
+    - key
+      timeout [sec]: Set waiting time for updating /battery_state topic
+
+  ;; If /battery_state is subscribed before /battery_state becomes true
+  ;; while dock succeeds, the result will not be correct,
+  ;; so wait for a certain time when docking.
+  """
+  (let ((msg nil)
+        (is-charging nil)
+        (duration 0)
+        (start-time (send (ros::time-now) :sec)))
+    (while (and (< duration timeout) (not is-charging))
+      (setq msg (one-shot-subscribe "/battery_state" power_msgs::BatteryState))
+      (if msg (setq is-charging (send msg :is_charging)))
+      (setq duration (- (send (ros::time-now) :sec) start-time)))))
```

Comment:
It would be good to know the exit status of `(wait-until-is-charging)` (charging or timeout)

Reply:
Thank you for reviewing.
I support this in [`8384d01` (#264)](https://github.com/knorth55/jsk_robot/pull/264/commits/8384d019c3c7955e1e991e425a6a151711e75867)

Reply:
Nice!